dask_jobqueue/jobqueue.yaml

```diff
@@ -10,6 +10,7 @@ jobqueue:
     interface: null             # Network interface to use like eth0 or ib0
     death-timeout: 60           # Number of seconds to wait if a worker can not find a scheduler
     local-directory: null       # Location of fast local storage like /scratch or $TMPDIR
+    shared-directory: "~"       # Shared directory between scheduler and worker
```

Comment:
I hope this is OK. It could also default to null, however, this default should work for most users. If a cluster has no shared directory it should be set to null though manually.

Reply:
There is pros and cons. 
for null:
- any user who wants to use security without thinking needs to set this
+ it is a safe default and will make it easier to find errors if it does not work

for "~":
+ the home directory is probably a sensible default that 90% of the cluster users
+ if a user wants to just use security together with #527 it will work out of the box 
- if things go wrong here it gets difficult

So together with a verbose warning/error message I feel confortable to set this to `null` particularly since adding "shared_directory="~", security=True" is not that much of an overhead. It makes sense to adds docs as part of #527 . I guess better safe than sorry.

Reply:
For me, this config option is more like `log_directory`. And I would put it to null by default, and in the code just try the current directory (with a warning?) as a base path for writing security temp files.

This way, I'd think than in most setups this should work without setting something (you usually run your scripts from a shared directory, HOME dir or an HPC shared file system). But I might be wrong here.

Reply:
OK with me if we do the renaming above. For a shared directory I would have though "null" means that there is none. I'll fix once the name is decided.

Reply:
We have to make it clear somewhere in the comment (or warning) that a null shared directory default to current directory if needed.